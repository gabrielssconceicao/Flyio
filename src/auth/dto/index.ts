import { LoginDto } from './login.dto';
import { LoginResponseDto } from './login-response.dto';
import { RefreshTokenDto } from './refresh-token.dto';
import { TokenPayloadDto } from './token-payload.dto';

export { LoginDto, LoginResponseDto, RefreshTokenDto, TokenPayloadDto };
