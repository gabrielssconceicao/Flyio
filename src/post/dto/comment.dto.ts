import { ApiProperty } from '@nestjs/swagger';

class UserComment {
  @ApiProperty({ example: 'John Doe', description: 'User name' })
  name: string;

  @ApiProperty({ example: 'jdoe', description: 'User username' })
  username: string;

  @ApiProperty({
    example: 'https://example.com/profile.jpg',
    description: 'User profile image URL',
  })
  profileImg: string | null;
}

export class CommentDto {
  @ApiProperty({ example: 'd-43-5df-df4', description: 'Comment identifier' })
  id: string;

  @ApiProperty({ example: 'Hello world', description: 'Comment content' })
  content: string;

  @ApiProperty({ example: '2023-03-15 14:30:00', description: 'Comment date' })
  createdAt: Date;

  @ApiProperty({ type: UserComment, description: 'Comment creator' })
  user: UserComment;
}
