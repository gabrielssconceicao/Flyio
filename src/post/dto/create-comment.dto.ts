import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MaxLength, MinLength } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class CreateCommentDto {
  @ApiProperty({
    example: 'Hello world',
    description: 'Comment text',
    minLength: 1,
    maxLength: 255,
  })
  @IsNotEmpty({ message: dtoErrorMessages('Content', DtoErrorType.REQUIRED) })
  @IsString({ message: dtoErrorMessages('Content', DtoErrorType.IS_STRING) })
  @MinLength(1, {
    message: dtoErrorMessages('Content', DtoErrorType.MIN_LENGTH, 1),
  })
  @MaxLength(255, {
    message: dtoErrorMessages('Content', DtoErrorType.MAX_LENGTH, 255),
  })
  content: string;
}
