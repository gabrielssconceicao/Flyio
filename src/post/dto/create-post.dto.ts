import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MinLength, MaxLength } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class CreatePostDto {
  @ApiProperty({
    example: 'Hello world',
    description: 'Post text',
    minLength: 1,
    maxLength: 255,
  })
  @IsNotEmpty({ message: dtoErrorMessages('Text', DtoErrorType.REQUIRED) })
  @IsString({ message: dtoErrorMessages('Text', DtoErrorType.IS_STRING) })
  @MinLength(1, {
    message: dtoErrorMessages('Text', DtoErrorType.MIN_LENGTH, 1),
  })
  @MaxLength(255, {
    message: dtoErrorMessages('Text', DtoErrorType.MAX_LENGTH, 255),
  })
  text: string;
}
