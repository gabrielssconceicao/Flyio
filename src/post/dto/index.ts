import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
