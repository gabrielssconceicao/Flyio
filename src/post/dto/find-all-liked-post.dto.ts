import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

export class FindAllLikedPostsResponseDto extends FindAllResponseDto<PostEntity> {
  @ApiProperty({
    type: [PostEntity],
    description: 'List of posts liked by the user',
  })
  items: PostEntity[];
}
