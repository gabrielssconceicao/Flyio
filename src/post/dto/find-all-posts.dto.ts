import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

export class FindAllUsersResponseDto extends FindAllResponseDto<PostEntity> {
  @ApiProperty({
    type: [PostEntity],
    description: 'List of posts of all users',
  })
  items: PostEntity[];
}
