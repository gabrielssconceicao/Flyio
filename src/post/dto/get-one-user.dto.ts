import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { CommentDto } from './comment.dto';

export class GetOnePostDto extends PostEntity {
  @ApiProperty({
    isArray: true,
    description: 'List of comments',
  })
  comments: CommentDto[];
}
