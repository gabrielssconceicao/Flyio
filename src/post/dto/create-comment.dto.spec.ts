import { validate } from 'class-validator';
import { CreateCommentDto } from './create-comment.dto';
import { formatErrors } from '../../common/utils/dto-format.errors';

describe('<CreateCommentDto>', () => {
  let dto: CreateCommentDto;
  beforeEach(() => {
    dto = new CreateCommentDto();
  });
  it('should validate a DTO', async () => {
    dto.content = 'Hello world';
    const result = await validate(dto);
    expect(result.length).toBe(0);
  });

  it('should fail if content is empty', async () => {
    const result = await validate(dto);
    expect(result.length).toBe(1);
    expect(result[0].property).toBe('content');
    expect(formatErrors(result)).toMatchSnapshot();
  });
  it('should fail if content is too long', async () => {
    dto.content = 'A'.repeat(256);
    const result = await validate(dto);
    expect(result.length).toBe(1);
    expect(result[0].property).toBe('content');
    expect(formatErrors(result)).toMatchSnapshot();
  });
});
