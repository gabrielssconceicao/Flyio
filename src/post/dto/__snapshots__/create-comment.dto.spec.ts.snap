// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateCommentDto> should fail if content is empty 1`] = `
[
  {
    "constraints": {
      "isNotEmpty": "Content is required",
      "isString": "Content must be a string",
      "maxLength": "Content must be at most 255 characters long",
      "minLength": "Content must be at least 1 characters long",
    },
    "property": "content",
    "value": undefined,
  },
]
`;

exports[`<CreateCommentDto> should fail if content is too long 1`] = `
[
  {
    "constraints": {
      "maxLength": "Content must be at most 255 characters long",
    },
    "property": "content",
    "value": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
  },
]
`;
